src/google/protobuf/compiler/csharp/csharp_map_field.cc

```diff
@@ -94,9 +94,19 @@ void MapFieldGenerator::GenerateMergingCode(io::Printer* printer) {
 }
 
 void MapFieldGenerator::GenerateParsingCode(io::Printer* printer) {
-  printer->Print(
-    variables_,
-    "$name$_.AddEntriesFrom(ref input, _map_$name$_codec);\n");
+  GenerateParsingCode(printer, true);
+}
+
+void MapFieldGenerator::GenerateParsingCode(io::Printer* printer, bool use_parse_context) {
+  if (use_parse_context) {
```

Comment:
Possibly use a conditional operator here instead?

```
printer->Print(
  variables_,
  use_parse_context
  ? "$name$_.AddEntriesFrom(ref input, _map_$name$_codec);\n"
  : "$name$_.AddEntriesFrom(input, _map_$name$_codec);\n");
```

That has the benefit that the two versions are on consecutive lines, making the difference obvious.

Reply:
(There are a few others, too.)